### ISSUE TITLE
[Suggestion] Glass paint (material paint)

### ISSUE THREAD

**Issue by Uncle-Ulty** [tsunamayo/Starship-EVO#4882]
A recurrent request is MORE GLASS BRICKS.

Glass bricks are cool, are useful to make pretty stuff, and allow us to see the beautiful landscape of Starship EVO.

However, tons of new glass bricks make the codex full, confused, and kinda ugly...
So, instead of adding new glass bricks, and creating tons of new bricks, making the codex looks crowded and confused. Would it be possible to add "glass material paint" to make some bricks transparent?

with this material paint, we could transform most of the current bricks into glass versions. The transparency color would be settled using standard color. This "glass paint" would be exclusive to painting bricks, not decals.

Is it possible?

**Comment by Kaiser-Indrasil** [comment]
I second this, brilliant idea!

Avorion made something really similar - it has the same set of blocks (including 6th form :P ) across light armor, heavy armor, glow blocks, holographic blocks, and glass blocks.

**Comment by ProPeach** [comment]
That'd pretty much solve the quest for new glass shapes, except the canopy hepta. But but considering there is the hull arch and hollow tetra just like the canopy version, if there was a hollow hull hepta plus this suggestion would work out great at tidying up the codex

**Comment by Uncle-Ulty** [comment]
> That'd pretty much solve the quest for new glass shapes, except the canopy hepta. But but considering there is the hull arch and hollow tetra just like the canopy version, if there was a hollow hull hepta plus this suggestion would work out great at tidying up the codex ![image](https://user-images.githubusercontent.com/27857711/186264474-fbacaf6f-6a98-4873-b486-fb9ea283e286.png)

I need this brick... like... on every ship... you know, my ships are like a fancy chandelier, with lots of glass and ornamental details.

**Comment by ProPeach** [comment]
Yeah 100%, it's absolutely essential for anything more than the simplest of canopy shapes.
Although having said that I do understand that this shape might have some weird issues when you stretch it, I guess that's why it hasn't been included yet if it's gonna take more dev time to solve. Those triangle faces on each corner will stretch into weird wedges if you stretch it using the current system which makes it not match the other glass shapes

**Comment by ProPeach** [comment]
Ok, last shape suggestion, I swear...
![image](https://user-images.githubusercontent.com/27857711/188277876-0d89b8bd-f80a-4775-ab1b-83cc28c6d3b9.png)
![image](https://user-images.githubusercontent.com/27857711/188277847-a84b24de-327d-48f2-901d-cbb0b4ee8b98.png)
![image](https://user-images.githubusercontent.com/27857711/188277900-5a3a83b1-4328-42cc-9591-aca0261b9c72.png)

**Comment by Uncle-Ulty** [comment]
I'd love to have this. It would look awesome for station observation deck dome.

**Comment by tsunamayo** [comment]
@ProPeach Hi, could you send me those shapes, I wanted to do another pass on glass and other stuff now that we have variants. I feel much more comfortable to add more bricks ^^
Regarding the suggestion, nope I cant do this, that too complex as transparent require another pass.

**Comment by ProPeach** [comment]
> @ProPeach Hi, could you send me those shapes, I wanted to do another pass on glass and other stuff now that we have variants. I feel much more comfortable to add more bricks ^^

[SEVO Glass Shapes.zip](https://github.com/tsunamayo/Starship-EVO/files/9936712/SEVO.Glass.Shapes.zip)
Sure, here you go! As before the hepta might actually be a little challenging, but I could be misreading the stretch. Excited to hear that about the possibility for other variants, seems players always love more bricks to play with heh

**Comment by tsunamayo** [comment]
I wanted to make the rounded cockpit too. I also got an idea to make side less windows, so you can design cockpit more freely. The front face and backface would only be connected minimaly through the corner of the brick, so there would not be any distracting side when two brick join.

**Comment by ProPeach** [comment]
Oh hmmm, so the faces which usually cause the Z-fighting simply wouldn't be there in the model? That'd be pretty interesting to see! Although it would add another variant of each brick, but that's not such a big issue anymore

**Comment by Kaiser-Indrasil** [comment]
@tsunamayo I love the idea of the sideless windows! Just look at smets' canopy and imagine how much cooler would it look if the game only rendered the front and back faces without rendering the glass faces that are touching other side glass faces:

![image](https://user-images.githubusercontent.com/63756057/200080556-04120bd4-05b6-42be-829d-e394e6a3ec3f.png)

**Comment by JRL101** [comment]
I was attempting to fill in a canopy when i realised i dont have the shapes for it. on a small scale. 
![image](https://user-images.githubusercontent.com/33255536/202861340-b7cfcf5b-12ac-428a-bec5-dd959153bcd4.png)
This is the tetra piece for glass, and this is as small as it goes. Theres also not any hepta canopy blocks.

Why are the glass blocks different from the regular blocks? I know canopy blocks are meant to be "hollow" inside, why not the same blocks but with a "glass shader" then have canopy blocks that can ALSO be have the glass or non glass shaders applied?
Im sure theres something stopping it for being a thing. like the glass blocks need a special block because they have other code to make them cull touching faces etc.

**Comment by ProPeach** [comment]
> I was attempting to fill in a canopy when i realised i dont have the shapes for it. on a small scale.
image
This is the tetra piece for glass, and this is as small as it goes

@JRL101, idk if you notices but you can get smaller scales than that to play with. Hit F1, you can reduce the minimum scale down to 1/8th which halves the size of the canopy piece in that picture

**Comment by JRL101** [comment]
> > I was attempting to fill in a canopy when i realised i dont have the shapes for it. on a small scale.
> > image
> > This is the tetra piece for glass, and this is as small as it goes
> 
> @JRL101, idk if you notices but you can get smaller scales than that to play with. Hit F1, you can reduce the minimum scale down to 1/8th which halves the size of the canopy piece in that picture

Its still the wrong scale?
